codegenerator/cli/templates/static/codegen/src/LoadLayer.res

```diff
@@ -82,166 +76,238 @@ module LoadActionMap = {
   }
 }
 
-type t = dict<ref<LoadActionMap.t<Entities.internalEntity>>>
-
-let make = (~config: Config.t) => {
-  let maps = Js.Dict.empty()
-  config.entities->Js.Array2.forEach(entityMod => {
-    let module(Entity) = entityMod
-    maps->Js.Dict.set(Entity.key, ref(LoadActionMap.empty()))
-  })
-  maps
+type rec t = {
+  mutable byEntity: dict<LoadActionMap.t>,
+  mutable isScheduled: bool,
+  mutable inMemoryStore: InMemoryStore.t,
```

Comment:
Why is inMemoryStore moved as state here?

The inMemory store is not attached to a load layer, it's referenced since other contexts need to access it.

Reply:
I'm sure at the end of the day it works out since js will pass objects by reference and all operations on the in memory store are mutations. But I think we should avoid these patterns if possible. If we are referencing a mutable object it should be passed via function parameter IMO

Reply:
Agree, it'll make things clearer and more maintainable 👍